aten/src/ATen/native/cuda/int4mm.cu

```diff
@@ -125,9 +127,38 @@ inline __host__ __device__ uint32_t getAlignmentRoundUp(const void* p) {
   return diff == 0 ? 0 : uint32_t(Align) - diff;
 }
 
+#if defined(USE_ROCM)
+// TODO: Support RDNA
+constexpr int32_t kWarpSize = 64;
+
+template<typename T, uint32_t Rank>
+using VecT = T __attribute__((ext_vector_type(Rank)));
+
+static bool isCDNA2orLater(int index) {
+    hipDeviceProp_t* prop = at::cuda::getDeviceProperties(index);
+    std::string device_arch = prop->gcnArchName;
+    static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942"};
+    for (std::string arch : archs) {
+        size_t substring = device_arch.find(arch);
+        if (substring != std::string::npos) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
This function seems identical to `isSupportedHipLtROCmArch`, perhaps they could be moved to a shared header and renamed ot something common?
https://github.com/pytorch/pytorch/blob/b5fdbc1a9fb94e80f0addd836fee358d2446ea88/aten/src/ATen/native/cuda/Blas.cpp#L207-L220

Reply:
Sure, will clean this up in a follow-up PR